flytekit/remote/executions.py

```diff
@@ -132,13 +140,58 @@ def is_done(self) -> bool:
         }
 
     @classmethod
-    def promote_from_model(cls, base_model: execution_models.Execution) -> "FlyteWorkflowExecution":
+    def promote_from_model(
+        cls, base_model: execution_models.Execution, remote: Optional["FlyteRemote"] = None
+    ) -> "FlyteWorkflowExecution":
         return cls(
+            remote=remote,
             closure=base_model.closure,
             id=base_model.id,
             spec=base_model.spec,
         )
 
+    def sync(self, sync_nodes: bool = False) -> "FlyteWorkflowExecution":
+        """
+        Sync the state of the current execution and returns a new object with the updated state.
+        """
+        if self._remote is None:
+            raise user_exceptions.FlyteAssertion("Cannot sync without a remote")
+        return self._remote.sync_execution(self, sync_nodes=sync_nodes)
+
+    def wait(
+        self,
+        timeout: Optional[Union[timedelta, int]] = None,
+        poll_interval: Optional[Union[timedelta, int]] = None,
+        sync_nodes: bool = True,
+    ) -> "FlyteWorkflowExecution":
+        """
+        Wait for the execution to complete. This is a blocking call.
+        :param timeout: The maximum amount of time to wait for the execution to complete. It can be a timedelta or
+         a duration in seconds as int.
+        :param poll_interval: The amount of time to wait between polling the state of the execution. It can be a
+            timedelta or a duration in seconds as int.
+        """
+        if self._remote is None:
+            raise user_exceptions.FlyteAssertion("Cannot wait without a remote")
+        if poll_interval is not None and not isinstance(poll_interval, timedelta):
```

Comment:
Can we update `FlyteRemote.wait` to support int so that `FlyteWorkflowExecution` can just dispatch to it?

Reply:
I guess I can, let me do that

Reply:
I didn't have the access right to push to the PR...